luna/gateware/usb/usb2/control.py

```diff
@@ -135,7 +133,7 @@ def elaborate(self, platform):
         if len(self._request_handlers) > 0:
             # Emit a stall when the request is not taken care of by the registered request handlers
             handle_conditions = [handler.handled for handler in self._request_handlers]
-            stall_condition = lambda setup: reduce(operator.and_, (~cond(setup) for cond in handle_conditions))
+            stall_condition = lambda setup: sum((cond(setup) for cond in handle_conditions)) != 1
         else:
             stall_condition = lambda setup: True
```

Comment:
The outer if/else could be removed now, because in the case where there are no handlers, the sum will be zero.